src/main/java/use_cases/DataAccess.java

```diff
@@ -1,10 +1,11 @@
 package use_cases;
 
+import entities.GameObserver;
 import entities.HumanPlayer;
 
 import java.io.IOException;
 
-public interface DataAccess {
+public interface DataAccess extends GameObserver {
```

Comment:
whats the reasoning behind making data access extend game observer? in my mind, we dont need to tangle the two interfaces. rather, we can just have CSVDatabase extend the observer interface. this shouldnt affect the code in a big way either

Reply:
It wouldn't affect the code but it's more of a CA thing. Essentially the goal is to have any DataAccess object update itself. If we didn't have the update() method in DataAccess, then for any future extensions such as SQL databases or anything else, the data access class for the isn't contractually obliged to update itself since the DataAccess interface doesn't specify that DataAccess object should update themselves. Also having CSVDatabase extend the observer interface directly would be having a driver/db class implement an entity interface, which would violate CA, there would also be another violation when update() in CSVDatabase uses entity classes without a use case layer interface between it.

Reply:
but CSVDatabase is already implementing entity classes. if you want to separate them, consider making another class that implements the observer and has a reference to a dataaccess instance. when that class is notified of a change in the game, it sees if that change is a win, and if it is, passes the appropriate data to the dataaccess

Reply:
That would definitely work and I'll try that, but I'd feel like we'd have too many classes for getting the Game information. I could try doing this with GameState though.

Reply:
we can figure out the specifics later